src/bindings/python/flux/queue.py

```diff
@@ -0,0 +1,345 @@
+        ):
+            self._resource_list = resources.copy_constraint(
+                {"properties": config["queues"][name]["requires"]}
+            )
+        else:
+            self._resource_list = resources
+
+    def __getattr__(self, attr):
+        return getattr(self._resource_list, attr)
+
+
+class QueueInfo:
+    """
+    Information for a single queue.
+
+    Attrs:
+        name (str): The queue name (empty string for anonymous queue)
+        is_default (bool): True if this is the default queue
+        enabled (bool): True if this queue is enabled
+        started (bool): True if this queue is started
+        resources (:obj:`QueueResources`): resources currently in this queue
+        defaults (:obj:`QueueDefaults`): defaults that apply to this queue
+        limits (:obj:`QueueLimits`): policy limits that apply to this queue
+    """
+
+    def __init__(self, name, config, resources, enabled, started, default):
+        self.name = name or ""
+        self.config = config
+        self.is_default = default
+        self.enabled = enabled
+        self.started = started
+        self.resources = QueueResources(name, resources, config)
+        self.defaults = QueueDefaults(
+            duration=parse_fsd(self.policy_default("duration"))
+        )
+        self.limits = QueueLimits(
+            min=QueueResourceCounts(
+                nnodes=self.size_limit("nnodes", False),
+                ncores=self.size_limit("ncores", False),
+                ngpus=self.size_limit("ngpus", False),
+            ),
+            max=QueueResourceCounts(
+                nnodes=self.size_limit("nnodes"),
+                ncores=self.size_limit("ncores"),
+                ngpus=self.size_limit("ngpus"),
+            ),
+            duration=parse_fsd(self.policy_system_limit("duration")),
+        )
+
+    def size_limit(self, key, maximum=True):
+        limit = maximum and "max" or "min"
+        try:
+            val = self.config["queues"][self.name]["policy"]["limits"]["job-size"][
+                limit
+            ][key]
+        except KeyError:
+            try:
+                val = self.config["policy"]["limits"]["job-size"][limit][key]
+            except KeyError:
+                val = math.inf if maximum else 0
+            if val < 0:
+                val = math.inf
+        return val
+
+    def policy_default(self, key, default="inf"):
+        try:
+            result = self.config["queues"][self.name]["policy"]["jobspec"]["defaults"][
+                "system"
+            ][key]
+        except KeyError:
+            try:
+                result = self.config["policy"]["jobspec"]["defaults"]["system"][key]
+            except KeyError:
+                result = default
+        return result
+
+    def policy_system_limit(self, key, default="inf"):
+        try:
+            result = self.config["queues"][self.name]["policy"]["limits"][key]
+        except KeyError:
+            try:
```

Comment:
I _think_ for the exception handling here, you could use `.get()` for the dictionary access and provide a default value in the case where the key-value access fails, and you won't need an inner `try-except` block here, i.e:

```python
result = self.config.get("policy").get("limits").get(key, default)
```

But this is probably no big deal either way.

Reply:
This suggestion also assumes that the previous keys (`"queues"`, `self.name`, `"policy"`, and `"limits"`) are all guaranteed to be present. Otherwise, an `AttributeError` will be raised. So, if those are not guaranteed to be present in the dictionary, sticking with the style you proposed probably makes more sense!

Reply:
They are not guaranteed to be present. I guess each `.get()` could return an empty dict, e.g.
```python
result = self.config.get("policy", {}).get("limits", {}).get(key, default)
```

However, this is copying existing code from `flux-queue.py` and it would probably be better to not change too much at once. We can come by and fix this in a future PR. Thanks!

BTW, I should have also mentioned that this PR is based on top of #6723, so you could have ignored the earlier commits from that PR. Sorry!

Reply:
No problem at all! Thanks for the clarification. This LGTM then!